Update ADR on SSO-Identity-Federation (#270)
SovereignCloudStack/standards: Standards/scs-0300-v1-requirements-for-sso-identity-federation.md

  - How would we allow SCS operators to choose?
- Do we need some kind of SWOT analysis to come to a decision?

## Decision

SCS wants to make use of an IdP as part of the reference implementation.
To move forward with topics of configuration and mapping of roles in a

> OAuth federated scenario as well as questions of token lifecycles etc across

Reviewer: ```suggestion
OAuth2 federation scenario as well as questions of token lifecycles etc. across
```

> the federation stack it makes sense to fokus on one IdP implementation at a

Reviewer: ```suggestion
the federation stack it makes sense to focus on one IdP implementation at a
```

> given time.

Reviewer: ```suggestion
given time. Both considered options seem to be potentially viable, but ultimately,
a decision should be made, even if there are no strict/strong reasons for
dismissing either option in particular.
```

Since Keycloak currently supports the OAuth 2.0 grants that SCS wants to make

> use of (e.g. Device Authorization Grant) and offers additional flexibility

Reviewer: As of a [recent version](https://github.com/zitadel/zitadel/releases/tag/v2.26.0) of Zitadel, Device Auth should be included as well. If (at the time of writing) there are still problems with it, let's spell it out more explicitly.

Author: I'd rather not do yet another round of research what Zitadel does and does not support "at the time of writing" (and then again "at time of review"), and what things need to be added by writing Zitadel actions. That what I've written was valid at the time the decision has been proposed this way.

Reviewer: I understand, this makes sense, especially regarding the Device Auth Grant.

My thinking (more so about the mentioning of LDAP) was that it may be better to not provide arguments about non-critical features at all, rather than deciding on claims about features that become outdated quickly - which (to the uninitiated reader) may prompt the idea of re-evaluation of the decision.

(Also, to me, "we have a working setup with X already" would already be a strong reason to decide for X, if X and Y do not outmatch each other considerably)

Anyway, that's just about wording; My PR-approval still stands either way.

Author: Ok, I get your point. Looking at https://github.com/zitadel/zitadel/discussions/1938 doesn't tell me that anything changed in Zitadel with respect to LDAP?

Reviewer: Seems to be a feature, now: https://zitadel.com/docs/guides/integrate/identity-providers/ldap

Author: Ah cool, I updated the ADR, so from my side it's fine now.

> e.g. by allowing LDAP to be used as backend, the projects current choice

Reviewer: ```suggestion
e.g. by allowing LDAP to be used as backend, the project's current choice
```

Reviewer: Also, Zitadel [supports LDAP](https://zitadel.com/docs/guides/integrate/identity-providers/ldap) as backend now, as well.
So this specific feature may be dropped as advantage of Keycloak, here.
